.github/workflows/tests.yml

```diff
@@ -144,12 +144,18 @@ jobs:
     # TODO: when actions supports using ${{env}} in job.*.if and not just job.*.steps.*.if, use this:
     #if: ${{ env.NIGHTLY }}
     # in the meantime, copy-paste this:
-    if: ${{ github.event_name == 'schedule' || github.event_name == 'push' || github.event_name == 'workflow_dispatch' }}
+    # if: ${{ github.event_name == 'schedule' || github.event_name == 'push' || github.event_name == 'workflow_dispatch' }}
     runs-on: ubuntu-20.04
     container: debian:9
     steps:
       - name: Dependencies
-        run: |  # NB: libgl1-mesa-dev is needed for PyQT testing (https://stackoverflow.com/q/33085297/7584115)
+        # NB: libgl1-mesa-dev is needed for PyQT testing (https://stackoverflow.com/q/33085297/7584115)
+        # NB: We echo "archive.debian.org" to address https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4105
+        run: |  
+          echo "deb http://archive.debian.org/debian/ stretch main non-free contrib" > /etc/apt/sources.list
+          echo "deb-src http://archive.debian.org/debian/ stretch main non-free contrib" >> /etc/apt/sources.list
+          echo "deb http://archive.debian.org/debian-security/ stretch/updates main non-free contrib" >> /etc/apt/sources.list
+          echo "deb-src http://archive.debian.org/debian-security/ stretch/updates main non-free contrib" >> /etc/apt/sources.list
```

Comment:
For these commands, I followed https://www.howtoforge.com/using-old-debian-versions-in-your-sources.list as a rough guide, but it would be nice to double-check if everything here is 100% necessary.

Reply:
Looking at my own `sources.list`, I see essentially the same four lines, so this seems fine to me.

But, a suggestion: if we literally just want to replace `deb.debian.org` and `security.debian.org` with `archive.debian.org`, why not do exactly that:
```sh
sed -i -e 's/\(deb\|security\)\.debian\.org/archive.debian.org/' /etc/apt/sources.list
```

Reply:
Good thinking! I feel like pattern-based techniques (regex, sed, etc.) are definitely a blind spot of mine. Too rarely do I consider them as a first option? 

The only thing I'm unsure about is whether the default entries for Debian 9 map to `archive` links. If [this link](https://www.geekersdigest.com/apt-etc-apt-sources-list-for-every-debian-release/) is to be trusted, then there is an additional source:

```diff
deb http://deb.debian.org/debian/ stretch main contrib non-free
+deb http://deb.debian.org/debian/ stretch-updates main contrib non-free
deb http://security.debian.org/ stretch/updates main contrib non-free
```

Checking http://archive.debian.org/debian/ for `stretch-updates` shows no such distribution:

![image](https://user-images.githubusercontent.com/16181459/234646563-21c89fcf-c8cd-45a4-adee-daa0d34776b3.png)

So, I have a feeling that the proposed `sed` command might result in a 404 not found. (But, nonetheless, I'd be happy to try it, as it's a much more elegant solution.)

Reply:
I wonder if `stretch-updates` is supposed to map to `stretch-proposed-updates`? :thinking: